pythran/optimizations/square.py

```diff
@@ -48,18 +50,35 @@ def visit_Module(self, node):
             node.body.insert(0, importIt)
         return node
 
+    def expand_pow(self, node, n):
+        if n == 0:
+            return ast.Num(1)
+        elif n == 1:
+            return node
+        else:
+            node_square = self.replace(node)
+            node_pow = self.expand_pow(node_square, n >> 1)
+            if n & 1:
+                return ast.BinOp(node_pow, ast.Mult(), copy.deepcopy(node))
+            else:
+                return node_pow
```

Comment:
Why did you implement it here instead of in the Pythonic runtime?
I agree that for now, this rewriting would certainly have better performances as it would totally avoid all parity comparison for exponent at runtime.
One the other side, It will not work for values not known at compile time and it would certainly not correctly use SIMD register. (The compiler will not known that for x^n, x is used n times and we don't need to unload it to reload the same value for the next matrix)